### PR TITLE
#82 refactor route handling

### LINKED ISSUE #82: Refactor route handling
- Refactor handling of routes with more than 2 slashes
- Stop server from throwing errors when bad routes are calles 
- fix api routes (for example `api/modelname/` is not working, but should display the same as `api/modelname`)

### PR DESCRIPTION
**Pull request by essal0815**
fixes #82

### CODE REVIEW

File: packages/RatPack.package/RPTestApp.class/instance/handleRequestForCaptureGroupsWithNumber.andWord..st [hunk 1]
```diff
@@ -0,0 +1,6 @@
+routes
+handleRequestForCaptureGroupsWithNumber: aNumber andWord: aWord
+	<get: '/captureGroups/number/([\d]+)/word/([\w]+)'>
```
Comment (richardebeling) [inline packages/RatPack.package/RPTestApp.class/instance/handleRequestForCaptureGroupsWithNumber.andWord..st]: the route could be just (\d+) and (\w+), you don't need the [] here afaik,

File: packages/RatPack.package/RPTestApp.class/instance/handleRequestForManySlashes.st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+routes
+handleRequestForManySlashes
+	<get: '/thisDoesExist/butDontTakeArguments/even/with/many/Slashes'>
```
Comment (richardebeling) [inline packages/RatPack.package/RPTestApp.class/instance/handleRequestForManySlashes.st]: doesn't

File: packages/RatPack.package/RPTests.class/instance/testCaptureGroups.st [hunk 3]
```diff
@@ -0,0 +1,6 @@
+test - routing
+testCaptureGroups
+	
+	RPTestHelper get: '/captureGroups/number/1/word/Test' on: app.
```
Comment (richardebeling) [inline packages/RatPack.package/RPTests.class/instance/testCaptureGroups.st]: 1 is a somewhat convenient number (or at least somehow the the canonic number that would come out if you asked a lazy developer: Say a number). I would suggest using something else than 0 or 1 as I think it has a higher probability to fail.

File: packages/RatPack.package/RPTests.class/instance/testNotFound.st [hunk 4]
```diff
@@ -4,4 +4,7 @@ testNotFound
 	result := RPTestHelper get: '/thisDoesNotExist' on: app.
 	self assert: (result contents contents includesSubString: 'Try this').
 	self assert: result status = #notFound.
-	self assert: result contentType = 'text/html'
\ No newline at end of file
+	self assert: result contentType = 'text/html'.
+	
+	RPTestHelper get: '/thisDoesNotExist/(.*)/andCaptureGroupsInURLDontMakeProblem' on: app.
```
Comment (richardebeling) [inline packages/RatPack.package/RPTests.class/instance/testNotFound.st]: AreNoProblem

File: packages/RatPack.package/RPTestApp.class/instance/handleRequestForManySlashes.st (hunk 2 above)
Comment (essal0815) [reply packages/RatPack.package/RPTestApp.class/instance/handleRequestForManySlashes.st]: No it does
I have one for both

File: packages/RatPack.package/RPTestApp.class/instance/handleRequestForCaptureGroupsWithNumber.andWord..st (hunk 1 above)
Comment (essal0815) [reply packages/RatPack.package/RPTestApp.class/instance/handleRequestForCaptureGroupsWithNumber.andWord..st]: Ok

File: packages/RatPack.package/RPTestApp.class/instance/handleRequestForManySlashes.st (hunk 2 above)
Comment (essal0815) [reply packages/RatPack.package/RPTestApp.class/instance/handleRequestForManySlashes.st]: Understood now what you meant ;)

File: packages/RatPack.package/RPTests.class/instance/testCaptureGroups.st (hunk 3 above)
Comment (essal0815) [reply packages/RatPack.package/RPTests.class/instance/testCaptureGroups.st]: Fair enough changed it to 123456789

**Comment by richardebeling** [comment]
A quick notice: Github has [Keywords](https://help.github.com/articles/closing-issues-using-keywords/) that connect pull requests and issues. When you write `fixes #issue` somewhere in the PR, the issue will be closed when the PR is accepted and github shows nice linkage between them. We should use that. I added the comment to your first post.

**Review by richardebeling — APPROVED** [review]

**Comment by essal0815** [comment]
If checks pass, shall I merge ?

**Comment by richardebeling** [comment]
From my side, yes

**Pull request merged**